src/Microsoft.AspNetCore.Sockets.Client/Connection.cs

```diff
@@ -183,34 +185,37 @@ public Connection(Uri url, ILoggerFactory loggerFactory)
             _logger.LogTrace("Ending receive loop");
         }
 
-        public Task<bool> SendAsync(byte[] data, MessageType type)
+        public Task<Exception> SendAsync(byte[] data, MessageType type)
```

Comment:
We don't want to return a faulted task because we don't want to throw from SendAsync (if the call is awaited). You can think of this Exception as a more sophisticated success/failure flag which you could rethrow. Or we can stick to bool as we had before because the user can get the exception that caused the failure in the Closed event (which is inevitable if SendAsync fails)